desktop/src/main.rs

```diff
@@ -67,3 +67,60 @@ fn main() {
 
 	event_loop.run_app(&mut app).unwrap();
 }
+
+async fn init_wgpu_context() -> WgpuContext {
```

Comment:
I don't think this is the right place to put this function. I'd generall prefer if the reusable logic was moved directly into the wgpu executor (e.g. such that it can also be used by the graphene cli etc.) and for the desktop specific parts I'd like to move those to a different file in order to not clutter up the `main.rs`

Reply:
Forgot about graphene cli, sure. Are we sure we want the exact same adapter selection code for both? what would you like the adapters to be addressed by? Is the index (that is stable for a specific system/hardware configuration), I also thought about hashing the adapters info and using 4 bytes as a id. I will implement the printing as fmt then.

Reply:
I would expose multiple function with different levels of granularity. We could have one function which does everything automagically, one which allows you to request the graphics api and one which gives you full manual control over the adapter selection. But I'd like to avoid constructign the context outside of the wgpu executor crate in case we in the future need to enable extra extensions etc. It would be a nightmare if we had to keep those in sync